sway/tree/container.c

```diff
@@ -1380,8 +1377,8 @@ void container_add_child(struct sway_container *parent,
 	container_for_each_child(child, set_workspace, NULL);
 	bool fullscreen = child->pending.fullscreen_mode != FULLSCREEN_NONE ||
 		parent->pending.fullscreen_mode != FULLSCREEN_NONE;
-	set_fullscreen_iterator(child, &fullscreen);
-	container_for_each_child(child, set_fullscreen_iterator, &fullscreen);
+	set_fullscreen(child, &fullscreen);
+	container_for_each_child(child, set_fullscreen, &fullscreen);
```

Comment:
i'm not sure what's going on here, so i didn't touch it yet

Reply:
Sway is spreading the fullscreen state to new children of the container. If we don't want children to inherit the fullscreen state, this part is no longer necessary.

I tried adding a new child to a fullscreen container in i3 and it refused with a "((null))" protocol error message, so I suppose we should just no longer allow this by just refusing the move command.